dali/python/nvidia/dali/plugin/numba/experimental/__init__.py

```diff
@@ -549,17 +550,30 @@ def _check_minimal_numba_version(throw: bool = True):
 
     @staticmethod
     def _check_cuda_compatibility(throw: bool = True):
-        toolkit_version = cuda.runtime.get_version()
-        driver_version = cuda.driver.driver.get_version()
+        toolkit_version = nb_cuda.runtime.get_version()
+        driver_version = nb_cuda.driver.driver.get_version()
 
         # numba_cuda should handle the compatibility between toolkit and driver versions
         # otherwise check if if driver and runtime matches, or if the last working numba version
         # matches the driver for CUDA 12
-        if not importlib.util.find_spec("numba_cuda") and (
+        try:
+            # try importing cuda.core as it can be used later to check the compatibility
+            # if is okay to fail as it may not be installed, the check later can handle this
```

Comment:
Nitpick:
```suggestion
            # try importing cuda.core as it can be used later to check the compatibility
            # it is okay to fail as it may not be installed, the check later can handle this
```